crates/polars-ops/src/frame/pivot/mod.rs

```diff
@@ -192,112 +192,133 @@ fn pivot_impl(
 
     let mut final_cols = vec![];
 
+    let mut column_column_name = columns[0].clone();
+    let mut binding = pivot_df.clone();
+    let pivot_df: &DataFrame = if columns.len() > 1 {
+        let fields = columns
+            .iter()
+            .map(|s| pivot_df.column(s).unwrap().clone())
+            .collect::<Vec<_>>();
+        column_column_name = format!("{{\"{}\"}}", columns.join("\",\""));
+        if pivot_df
+            .get_column_names()
+            .contains(&column_column_name.as_str())
+        {
+            polars_bail!(ComputeError: "cannot use column name {column_column_name} that \
+            already exists in the DataFrame. Please rename it prior to calling `pivot`.")
+        }
```

Comment:
this is very unlikely to ever happen

someone would need to have:
- one column called `'{"a","b"}'`, another called `'a'`, and another called `'b'`
- to pivot using `columns=['a', 'b']`

nonetheless, if it does happen, best to error IMO

Reply:
I think we shouldn't need error code for obscure but unlikely column names, or else the codebase would be littered with checks like this all over the place. Won't we just end up panicking elsewhere regardless?

Reply:
the problem is that this wouldn't panic, but silently lead to wrong results